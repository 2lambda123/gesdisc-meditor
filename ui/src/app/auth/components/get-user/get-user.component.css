h3 {
  font-family: Roboto, "Helvetica Neue", sans-serif;
  font-size: 24px;
  color: grey;
  text-align: center;
}