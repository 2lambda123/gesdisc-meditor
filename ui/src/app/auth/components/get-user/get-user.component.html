<div fxLayout="row" fxLayout.lt-lg="column" fxLayoutAlign="center center" fxFill>
  <h3>Please wait while we redirect you back to the requested page!</h3>
</div>