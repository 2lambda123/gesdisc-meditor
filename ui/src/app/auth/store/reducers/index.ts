import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as fromAuth from './auth';

export interface AuthState {
  status: fromAuth.State;
}


export const reducers = {
  status: fromAuth.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(
  selectAuthStatusState, 
  fromAuth.getUser
);