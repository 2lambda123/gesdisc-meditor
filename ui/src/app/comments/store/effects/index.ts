import { CommentsEffects } from './comments.effects';

export const effects: any[] = [CommentsEffects];

export * from './comments.effects';