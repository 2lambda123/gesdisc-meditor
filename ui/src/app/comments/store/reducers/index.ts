import {
	createFeatureSelector,
	ActionReducerMap,
} from '@ngrx/store';
import * as fromComments from './comments.reducer';

export interface CommentsState {
	comments: fromComments.State;
}

export const reducers: ActionReducerMap<CommentsState> = {
	comments: fromComments.reducer
};

export const getCommentsState = createFeatureSelector<CommentsState>('comments');



