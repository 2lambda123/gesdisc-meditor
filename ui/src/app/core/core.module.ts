import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
