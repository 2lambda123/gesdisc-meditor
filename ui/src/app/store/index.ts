export { DocumentState } from './document/document.state';
export { ModelState } from './model/model.state';
export { NotificationState } from './notification/notification.state';
export { AuthState } from './auth/auth.state';
export { RouterState } from './router/router.state';
