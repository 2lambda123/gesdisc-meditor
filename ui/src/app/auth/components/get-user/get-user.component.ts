import { Component, OnInit } from '@angular/core';

import { Store } from '@ngrx/store';
import * as fromApp from '../../../store';
import * as Auth from '../../store';

@Component({
  selector: 'med-get-user',
  templateUrl: './get-user.component.html',
  styleUrls: ['./get-user.component.css']
})
export class GetUserComponent implements OnInit {

  constructor( private store: Store<fromApp.AppState>) { }

  ngOnInit() {
    this.store.dispatch(new Auth.GetUser());
  }  
}
