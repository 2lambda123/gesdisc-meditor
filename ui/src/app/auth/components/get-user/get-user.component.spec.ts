import { ComponentFixture, TestBed, ComponentFixtureAutoDetect} from '@angular/core/testing';
import { Store, NgxsModule } from '@ngxs/store';
import { GetUserComponent } from './get-user.component';

describe('GetUserComponent', () => {
	let component: GetUserComponent;
	let fixture: ComponentFixture<GetUserComponent>;
	let store: Store;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NgxsModule.forRoot()
			],
			declarations: [ GetUserComponent ],
			providers: [{ provide: ComponentFixtureAutoDetect, useValue: true }]
		});

		const initialState = {};
		fixture = TestBed.createComponent(GetUserComponent);
		component = fixture.componentInstance;
		store = TestBed.get(Store);
		store.reset(initialState);
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(fixture).toMatchSnapshot();
	});
});
