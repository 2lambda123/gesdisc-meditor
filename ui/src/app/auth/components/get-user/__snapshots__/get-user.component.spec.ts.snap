// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetUserComponent should create 1`] = `
<med-get-user
  store={[Function Store]}
>
  <div
    fxfill=""
    fxlayout="row"
    fxlayout.lt-lg="column"
    fxlayoutalign="center center"
  >
    <h3>
      Please wait while we redirecting you back to requested page!
    </h3>
  </div>
</med-get-user>
`;
